src/map/entities/mobentity.cpp

```diff
@@ -683,30 +683,183 @@ void CMobEntity::DistributeRewards()
     }
 }
 
+// Return the list of seals that can drop based on the mob's level.
+// Rules:
+// - Mob  < 50: Beastmen's Seal
+// - Mob >= 50: Beastmen's Seal, Kindred's Seal
+// - Mob >= 70: Beastmen's Seal, Kindred's Seal, Kindred's Crest
+// - Mob >= 80: Beastmen's Seal, Kindred's Seal, Kindred's Crest, High Kindred's Crest
+// If Abyssea is not enabled, pool is limited to Beastmen's Seal and Kindred's Seal.
+auto CMobEntity::GetEligibleSeals() -> std::vector<uint16>
+{
+    constexpr uint16 BEASTMENS_SEAL      = 1126;
+    constexpr uint16 KINDREDS_SEAL       = 1127;
+    constexpr uint16 KINDREDS_CREST      = 2955;
+    constexpr uint16 HIGH_KINDREDS_CREST = 2956;
+
+    if (GetMLevel() >= 80 && luautils::IsContentEnabled("ABYSSEA"))
+    {
+        return { BEASTMENS_SEAL, KINDREDS_SEAL, KINDREDS_CREST, HIGH_KINDREDS_CREST };
+    }
+
+    if (GetMLevel() >= 70 && luautils::IsContentEnabled("ABYSSEA"))
+    {
+        return { BEASTMENS_SEAL, KINDREDS_SEAL, KINDREDS_CREST };
+    }
+
+    if (GetMLevel() >= 50)
+    {
+        return { BEASTMENS_SEAL, KINDREDS_SEAL };
+    }
+
+    return { BEASTMENS_SEAL };
+}
+
+// Return the list of Geode and Avatarites that can drop based on the mob's level.
+// Rules:
+// - Mob >= 50: Geodes of matching weather/day can drop. Weather takes priority.
+// - Mob >= 80: Avatarites of matching weather/day can also drop. Weather takes priority.
+auto CMobEntity::GetEligibleGeodes() -> std::vector<uint16>
+{
+    constexpr uint16 FLAME_GEODE   = 3297;
+    constexpr uint16 SNOW_GEODE    = 3298;
+    constexpr uint16 BREEZE_GEODE  = 3299;
+    constexpr uint16 SOIL_GEODE    = 3300;
+    constexpr uint16 THUNDER_GEODE = 3301;
+    constexpr uint16 AQUA_GEODE    = 3302;
+    constexpr uint16 LIGHT_GEODE   = 3303;
+    constexpr uint16 SHADOW_GEODE  = 3304;
+
+    constexpr uint16 IFRITITE  = 3520;
+    constexpr uint16 SHIVITE   = 3521;
+    constexpr uint16 GARUDITE  = 3522;
+    constexpr uint16 TITANITE  = 3523;
+    constexpr uint16 RAMUITE   = 3524;
+    constexpr uint16 LEVIATITE = 3525;
+    constexpr uint16 CARBITE   = 3526;
+    constexpr uint16 FENRITE   = 3527;
+
+    // clang-format off
+    std::map<uint8, uint16> geodeMap = {
```

Comment:
These maps are being rebuilt every time `GetEligibleGeodes` is called, so move them up into that other namespace